### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by modernpacifist**

### CODE REVIEW

**Comment by Massugunaen** [comment]
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/PIL/ImageFile.py", line 167, in load
    seek = self.load_seek
AttributeError: 'JpegImageFile' object has no attribute 'load_seek'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/modernpacifist/Documents/GitHubProjects/geophoto/main.py", line 34, in <module>
    imglib.saving_img(image, path, size)
  File "/home/modernpacifist/Documents/GitHubProjects/geophoto/imglib.py", line 55, in saving_img
    rotated_image = image_obj.rotate(degrees)
  File "/usr/lib/python3/dist-packages/PIL/Image.py", line 2009, in rotate
    return self.transform((w, h), AFFINE, matrix, resample, fillcolor=fillcolor)
  File "/usr/lib/python3/dist-packages/PIL/Image.py", line 2310, in transform
    (0, 0) + size, self, method, data, resample, fillcolor is None
  File "/usr/lib/python3/dist-packages/PIL/Image.py", line 2379, in __transformer
    image.load()
  File "/usr/lib/python3/dist-packages/PIL/ImageFile.py", line 170, in load
    seek = self.fp.seek
AttributeError: 'NoneType' object has no attribute 'seek'

**Comment by modernpacifist** [comment]
```
/usr/bin/python3.7 /home/modernpacifist/Documents/GitHubProjects/geophoto/main.py /home/modernpacifist/jpg/ /home/modernpacifist/jpg1/ google.com 200
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/PIL/ImageFile.py", line 167, in load
    seek = self.load_seek
AttributeError: 'JpegImageFile' object has no attribute 'load_seek'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/modernpacifist/Documents/GitHubProjects/geophoto/main.py", line 34, in <module>
    imglib.saving_img(image, path, size)
  File "/home/modernpacifist/Documents/GitHubProjects/geophoto/imglib.py", line 55, in saving_img
    rotated_image = image_obj.rotate(degrees)
  File "/usr/lib/python3/dist-packages/PIL/Image.py", line 2009, in rotate
    return self.transform((w, h), AFFINE, matrix, resample, fillcolor=fillcolor)
  File "/usr/lib/python3/dist-packages/PIL/Image.py", line 2310, in transform
    (0, 0) + size, self, method, data, resample, fillcolor is None
  File "/usr/lib/python3/dist-packages/PIL/Image.py", line 2379, in __transformer
    image.load()
  File "/usr/lib/python3/dist-packages/PIL/ImageFile.py", line 170, in load
    seek = self.fp.seek
AttributeError: 'NoneType' object has no attribute 'seek'

Process finished with exit code 1
```

**Comment by tmshv** [comment]
try [this](https://stackoverflow.com/a/12413650/1860097)

**Comment by modernpacifist** [comment]
@tmshv wow, rly cool hyperlink, how did you achieve it?

File: imglib.py [hunk 1]
```diff
@@ -49,13 +46,11 @@ def get_rotate(orientation):
...
     elif orientation == 8:
         return 90
-    return 0
 
 
-def operations_run1(required_img, saving_path, img_size):
-    image_obj = Image.open(required_img)
-    orientation = get_orientation(required_img)
-    degrees = get_rotate(orientation)
+def saving_img(image_obj, path, img_size):
```
Comment (tmshv) [inline imglib.py]: Wrong function name. This function do:
- image rotation
- image resizing
- saving

Rename it pozhalusto.

File: main.py [hunk 2]
```diff
@@ -15,23 +15,24 @@
...
 
-if os.path.exists(destination_folder):
-    pass
-else:
+if not os.path.exists(destination_folder):
     os.mkdir(destination_folder)
-
-os.chdir(source_folder)
+else:
+    pass
```
Comment (tmshv) [inline main.py]: remove this

**Review by tmshv — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Geophoto
+
```
Comment (tmshv) [inline README.md:2]: Describe here what this piece of software do

File: README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+## What this software does:
+1) Takes folder which contains huge amount of jpeg images
+2) Takes every image in order which contains geodata, images that do not contain geodata will be skipped
+3) Writes geodata to 'dataset.geojson' file, it will be automatically generated in ```<output_folder_path>```
+4) Resizes chosen images and saves them to ```<output_folder_path>```, this is made for preview purposes
+5) As a result you will have ```<output_folder_path>```, which contains all info you needed
+
```
Comment (tmshv) [inline README.md:9]: You forgot the *base URL prefix*

**Review by tmshv — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (Massugunaen) [reply README.md]: xD

**Comment by tmshv** [comment]
Where is `requirements.txt`

File: README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+1) Takes folder which contains huge amount of jpeg images - ```<source_folder_path>```
+2) Takes every image in order which contains geodata, images that do not contain geodata will be skipped
+3) Writes geodata to 'dataset.geojson' file, it will be automatically generated in ```<output_folder_path>``` directory
+4) Resizes chosen images, depending on the value you gave it in ```<size>``` parameter and saves them to ```<output_folder_path>```, this is made for comfortable preview purposes
+5) As a result you will have: ```<output_folder_path>```, which contains all required photos and .geojson file with metadata
+
+### Install:
+pip package manager is not available for this project yet, so you need to clone this repository:
+```
+$ git clone git@github.com:designunit/geophoto.git
```
Comment (tmshv) [inline README.md:13]: Replace with *https* url.

If I remember correctly, cloning with ssh url will require public key associated with this repository. Most likely, end user does not have one.

File: imglib.py [hunk 6]
```diff
@@ -49,13 +46,11 @@ def get_rotate(orientation):
...
     elif orientation == 8:
         return 90
-    return 0
 
 
-def operations_run1(required_img, saving_path, img_size):
-    image_obj = Image.open(required_img)
-    orientation = get_orientation(required_img)
-    degrees = get_rotate(orientation)
+def saving_modified_img(image_obj, path, img_size):
```
Comment (tmshv) [inline imglib.py:51]: Still unacceptable name

unacceptable-lemongrab-e1434739410519

**Review by tmshv — CHANGES_REQUESTED** [review]

File: imglib.py (hunk 6 above)
Comment (modernpacifist) [reply imglib.py:51]: Bro, you just posted cringe

File: README.md (hunk 5 above)
Comment (modernpacifist) [reply README.md:13]: I will test it and make changes based on the result.

Comment (modernpacifist) [reply README.md:13]: > Replace with _https_ url.
> 
> If I remember correctly, cloning with ssh url will require public key associated with this repository. Most likely, end user does not have one.

Test were run, it works the way it described.

**Review by tmshv — APPROVED** [review]

**Pull request merged**